src/main/java/graphql/execution/Async.java

```diff
@@ -19,29 +20,115 @@
 @SuppressWarnings("FutureReturnValueIgnored")
 public class Async {
 
-    @FunctionalInterface
-    public interface CFFactory<T, U> {
-        CompletableFuture<U> apply(T input, int index, List<U> previousResults);
+    public interface ParallelBuilder<T> {
+
+        void add(CompletableFuture<T> completableFuture);
+
+        CompletableFuture<List<T>> build();
     }
 
-    public static <U> CompletableFuture<List<U>> each(List<CompletableFuture<U>> futures) {
-        CompletableFuture<List<U>> overallResult = new CompletableFuture<>();
+    public static <T> ParallelBuilder<T> parallel(int expectedSize) {
```

Comment:
Naming is hard

But I would call this CombinedBuilder since its combines together `n` CFs and then allow to wait on them all the complete.

other names

* Async.AllOfBuilder
* Async.CombinedWaitBuilder